sdk/schemaregistry/Azure.Data.SchemaRegistry/src/SchemaRegistryClient.cs

```diff
@@ -0,0 +1,204 @@
+                return Response.FromValue(properties, response);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Gets the schema ID associated with the schema from the SchemaRegistry service.
+        /// </summary>
+        /// <param name="groupName">The name of the SchemaRegistry group.</param>
+        /// <param name="schemaName">The name of the schema.</param>
+        /// <param name="serializationType">The serialization format of the schema.</param>
+        /// <param name="schemaContent">The string representation of the schema's content.</param>
+        /// <param name="cancellationToken">The cancellation token for the operation.</param>
+        /// <returns>The properties of the schema, including the schema ID provided by the service.</returns>
+        public virtual async Task<Response<SchemaProperties>> GetSchemaIdAsync(string groupName, string schemaName, SerializationType serializationType, string schemaContent, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope("SchemaRegistryClient.GetSchema");
+            scope.Start();
+            try
+            {
+                var response = await RestClient.QueryIdByContentAsync(groupName, schemaName, serializationType, schemaContent, cancellationToken).ConfigureAwait(false);
+                var properties = new SchemaProperties(schemaContent, response.Headers.Location, response.Headers.XSchemaType, response.Headers.XSchemaId, response.Headers.XSchemaVersion);
+                return Response.FromValue(properties, response);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Gets the schema ID associated with the schema from the SchemaRegistry service.
+        /// </summary>
+        /// <param name="groupName">The name of the SchemaRegistry group.</param>
+        /// <param name="schemaName">The name of the schema.</param>
+        /// <param name="serializationType">The serialization format of the schema.</param>
+        /// <param name="schemaContent">The string representation of the schema's content.</param>
+        /// <param name="cancellationToken">The cancellation token for the operation.</param>
+        /// <returns>The properties of the schema, including the schema ID provided by the service.</returns>
+        public virtual Response<SchemaProperties> GetSchemaId(string groupName, string schemaName, SerializationType serializationType, string schemaContent, CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope("SchemaRegistryClient.GetSchema");
```

Comment:
I think the reason we don't use `nameof` is because this would require 2 `nameof` calls (one of the class name and one of the method name) and concatenating the strings together (extra memory and blah blah blah). Additionally, we don't differentiate in telemetry between async and non-async operations (for some reason), so I'd have to nameof of the sync method in both cases. However, you did find a bug here where I was using the old name of the method. I've fixed these broken ones. I've also put these into constants so at least they are declared in 1 place.

Reply:
We did this in SB because we have no sync methods so we couldn't do nameof on anything. I think the standard pattern is to use nameof here though if you have sync/async methods.